### PR TITLE
Cleans-up clusters after test run

### PR DESCRIPTION
**Pull request by chentex**
Signed-off-by: Vicente Zepeda Mas <vzepedam@redhat.com>

### Description

Cleans-up clusters after test run
- Clean-up after create-clusters test
- Clean-up after cluster-registration and re-registration tests

### CODE REVIEW

File: pkg/helpers/helpers.go
Comment (smalleni) [inline pkg/helpers/helpers.go:91]: This is a "soft delete" is it? So subsequent endpoint calls will just exclude these subscriptions? If that is the case, will we get the true endpoint performance since we will perhaps still execute some logic to skip archived clusters because we didn't hard delete?

Thoughts @chentex @dry923 ?

Comment (chentex) [reply pkg/helpers/helpers.go:91]: So the end-point it's already doing this logic of skipping `soft deleted` subscriptions, so in my opinion it will be the real performance of it.
If we did hit a performance issue here, it would mean the `DB` is missing an index in that particular field, and would be a good feedback for the team. (I would highly doubt that is missing)

Comment (chentex) [reply pkg/helpers/helpers.go:91]: Also I am still waiting on the review from Zohar, he might have more comments to do on this, since I asked him specifically to review this part.

Comment (smalleni) [reply pkg/helpers/helpers.go:91]: Sounds good!

Comment (markturansky) [reply pkg/helpers/helpers.go:91]: this should be DELETE on `/subscriptions/{id}`, which flags the Subscription as Deprovisioned and releases its resources.

Comment (markturansky) [reply pkg/helpers/helpers.go]: @chentex   if you're already deleting the cluster through CS, you shouldn't have to DELETE a subscription in AMS.  CS does that for you.

**Review by venkataanil — APPROVED** [review]
LGTM

**Pull request merged**
